### PR TITLE
[5.2] Check imagecolortransparent for validity

### PR DESCRIPTION
**Pull request by MacJoom**
Pull Request for Issue #44417 

### Summary of Changes

Check imagecolortransparent for validity in Image.php avoiding Error message:imagecolorsforindex(): Argument 2 ($color) is out of range 

### Testing Instructions

Put the provided dvr.gif in the images folder, enable thumbnail generation in the "FileSystem - Local" plugin.
Go to media images

### Actual result BEFORE applying this Pull Request

message:imagecolorsforindex(): Argument 2 ($color) is out of range 

### Expected result AFTER applying this Pull Request

Directory is displayed with thumbnails

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed
dvr

### CODE REVIEW

**Comment by brianteeman** [comment]
I have confirmed the bug with the provided image

When I apply the PR however media manager doesnt load successfully with browser console showing this
```

Uncaught SyntaxError: Unexpected token '<', "<br />
<b>"... is not valid JSON
    at JSON.parse (<anonymous>)
    at onSuccess (media-manager.min.js?70778f:20:17327)
    at n.onreadystatechange (core.min.js?2cb912:4:5125)

```

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on a9af0b603609fc67509c4d1799f2781216e95f5a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by brianteeman** [comment]
```
<br/>
<b>Warning</b>
:  imagecreatefrompng(): gd-png:  fatal libpng error: Read Error: truncated data in <b>D:\repos\j51\libraries\src\Image\Image.php</b>
on line <b>640</b>
<br/>
<br/>
<b>Warning</b>
:  imagecreatefrompng(): gd-png error: setjmp returns error condition in <b>D:\repos\j51\libraries\src\Image\Image.php</b>
on line <b>640</b>
<br/>
<br/>
<b>Warning</b>
:  imagecreatefrompng(): &quot;D:/repos/j51/images/ov002_03.png &quot;is not a valid PNG file in <b>D:\repos\j51\libraries\src\Image\Image.php</b>
on line <b>640</b>
<br/>{"success":true,"message":null,"messages":null,"data":[{"type":"dir","name":"banners","path":"local-images:\/banners","extension":"","size":"","mime_type":"","width":0,"height":0,"create_date":"2024-02-26T15:19:51+00:00","create_date_formatted":"2024-02-26 15:19","modified_date":"2024-08-07T14:21:14+00:00","modified_date_formatted":"2024-08-07 14:21","adapter":"local-images"},{"type":"dir","name":"headers","path":"local-images:\/headers","extension":"","size":"","mime_type":"","width":0,"height":0,"create_date":"2024-02-26T15:19:51+00:00","create_date_formatted":"2024-02-26 15:19","modified_date":"2024-02-26T15:19:51+00:00","modified_date_formatted":"2024-02-26 15:19","adapter":"local-images"},{"type":"dir","name":"sampledata","path":"local-images:\/sampledata","extension":"","size":"","mime_type":"","width":0,"height":0,"create_date":"2024-02-26T15:19:51+00:00","create_date_formatted":"2024-02-26 15:19","modified_date":"2024-07-22T18:12:11+00:00","modified_date_formatted":"2024-07-22 18:12","adapter":"local-images"},{"type":"dir","name":"test","path":"local-images:\/test","extension":"","size":"","mime_type":"","width":0,"height":0,"create_date":"2024-07-15T18:27:11+00:00","create_date_formatted":"2024-07-15 18:27","modified_date":"2024-07-16T15:44:15+00:00","modified_date_formatted":"2024-07-16 15:44","adapter":"local-images"},{"type":"dir","name":"timelapse","path":"local-images:\/timelapse","extension":"","size":"","mime_type":"","width":0,"height":0,"create_date":"2024-07-24T10:14:25+00:00","create_date_formatted":"2024-07-24 10:14","modified_date":"2024-07-24T10:14:55+00:00","modified_date_formatted":"2024-07-24 10:14","adapter":"local-images"},{"type":"file","name":"2 2.jpg","path":"local-images:\/2 2.jpg","extension":"jpg","size":10216,"mime_type":"image\/jpeg","width":500,"height":1000,"create_date":"2024-04-13T19:20:16+00:00","create_date_formatted":"2024-04-13 19:20","modified_date":"2024-09-29T12:10:13+00:00","modified_date_formatted":"2024-09-29 12:10","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/2%202.jpg","adapter":"local-images"},{"type":"file","name":"313913832_6078816912176680_6390327362732999404_n.jpg","path":"local-images:\/313913832_6078816912176680_6390327362732999404_n.jpg","extension":"jpg","size":82137,"mime_type":"image\/jpeg","width":526,"height":792,"create_date":"2024-07-22T16:29:10+00:00","create_date_formatted":"2024-07-22 16:29","modified_date":"2024-07-22T16:29:10+00:00","modified_date_formatted":"2024-07-22 16:29","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/313913832_6078816912176680_6390327362732999404_n.jpg","adapter":"local-images"},{"type":"file","name":"383550201-446fd6cf-d4a1-4d78-8343-3920bf6a1867.gif","path":"local-images:\/383550201-446fd6cf-d4a1-4d78-8343-3920bf6a1867.gif","extension":"gif","size":6813,"mime_type":"image\/gif","width":186,"height":192,"create_date":"2024-11-06T12:42:23+00:00","create_date_formatted":"2024-11-06 12:42","modified_date":"2024-11-06T12:42:31+00:00","modified_date_formatted":"2024-11-06 12:42","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/383550201-446fd6cf-d4a1-4d78-8343-3920bf6a1867.gif","adapter":"local-images"},{"type":"file","name":"451736287_10159729877296529_7926576536803486640_n.jpg","path":"local-images:\/451736287_10159729877296529_7926576536803486640_n.jpg","extension":"jpg","size":173814,"mime_type":"image\/jpeg","width":1200,"height":630,"create_date":"2024-07-22T16:31:00+00:00","create_date_formatted":"2024-07-22 16:31","modified_date":"2024-07-22T16:31:00+00:00","modified_date_formatted":"2024-07-22 16:31","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/451736287_10159729877296529_7926576536803486640_n.jpg","adapter":"local-images"},{"type":"file","name":"50809032_1875280872600938_647483276402884608_n.jpg","path":"local-images:\/50809032_1875280872600938_647483276402884608_n.jpg","extension":"jpg","size":43313,"mime_type":"image\/jpeg","width":675,"height":675,"create_date":"2024-07-22T16:35:28+00:00","create_date_formatted":"2024-07-22 16:35","modified_date":"2024-07-22T16:35:28+00:00","modified_date_formatted":"2024-07-22 16:35","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/50809032_1875280872600938_647483276402884608_n.jpg","adapter":"local-images"},{"type":"file","name":"80thtito.png","path":"local-images:\/80thtito.png","extension":"png","size":1306848,"mime_type":"image\/png","width":2000,"height":600,"create_date":"2024-07-23T10:41:47+00:00","create_date_formatted":"2024-07-23 10:41","modified_date":"2024-07-23T10:41:47+00:00","modified_date_formatted":"2024-07-23 10:41","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/80thtito.png","adapter":"local-images"},{"type":"file","name":"BrianTeeman-de-Ania-Gross.jpg","path":"local-images:\/BrianTeeman-de-Ania-Gross.jpg","extension":"jpg","size":345562,"mime_type":"image\/jpeg","width":1199,"height":848,"create_date":"2024-04-05T22:09:33+00:00","create_date_formatted":"2024-04-05 22:09","modified_date":"2024-04-05T22:23:18+00:00","modified_date_formatted":"2024-04-05 22:23","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/BrianTeeman-de-Ania-Gross.jpg","adapter":"local-images"},{"type":"file","name":"Dia\u0301logos.png","path":"local-images:\/Dia\u0301logos.png","extension":"png","size":3371175,"mime_type":"image\/png","width":1920,"height":3311,"create_date":"2024-07-19T22:13:56+00:00","create_date_formatted":"2024-07-19 22:13","modified_date":"2024-07-19T22:13:56+00:00","modified_date_formatted":"2024-07-19 22:13","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Dia\u0301logos.png","adapter":"local-images"},{"type":"file","name":"Red and White Simple Flat Illustrative Sunset Beach House  Club Logo 1.png","path":"local-images:\/Red and White Simple Flat Illustrative Sunset Beach House  Club Logo 1.png","extension":"png","size":14795,"mime_type":"image\/png","width":320,"height":100,"create_date":"2024-07-31T09:41:34+00:00","create_date_formatted":"2024-07-31 09:41","modified_date":"2024-07-31T09:41:34+00:00","modified_date_formatted":"2024-07-31 09:41","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Red%20and%20White%20Simple%20Flat%20Illustrative%20Sunset%20Beach%20House%20%20Club%20Logo%201.png","adapter":"local-images"},{"type":"file","name":"Red and White Simple Flat Illustrative Sunset Beach House  Club Logo 2.png","path":"local-images:\/Red and White Simple Flat Illustrative Sunset Beach House  Club Logo 2.png","extension":"png","size":7209,"mime_type":"image\/png","width":320,"height":100,"create_date":"2024-07-31T09:42:52+00:00","create_date_formatted":"2024-07-31 09:42","modified_date":"2024-08-12T10:07:09+00:00","modified_date_formatted":"2024-08-12 10:07","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Red%20and%20White%20Simple%20Flat%20Illustrative%20Sunset%20Beach%20House%20%20Club%20Logo%202.png","adapter":"local-images"},{"type":"file","name":"Red and White Simple Flat Illustrative Sunset Beach House  Club Logo.png","path":"local-images:\/Red and White Simple Flat Illustrative Sunset Beach House  Club Logo.png","extension":"png","size":9935,"mime_type":"image\/png","width":320,"height":100,"create_date":"2024-07-31T09:40:27+00:00","create_date_formatted":"2024-07-31 09:40","modified_date":"2024-07-31T09:40:27+00:00","modified_date_formatted":"2024-07-31 09:40","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Red%20and%20White%20Simple%20Flat%20Illustrative%20Sunset%20Beach%20House%20%20Club%20Logo.png","adapter":"local-images"},{"type":"file","name":"Screen Shot 2017-11-27 at 16.19.29.JPG","path":"local-images:\/Screen Shot 2017-11-27 at 16.19.29.JPG","extension":"JPG","size":77147,"mime_type":"image\/jpeg","width":577,"height":450,"create_date":"2024-04-20T15:58:13+00:00","create_date_formatted":"2024-04-20 15:58","modified_date":"2024-04-21T17:58:36+00:00","modified_date_formatted":"2024-04-21 17:58","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Screen%20Shot%202017-11-27%20at%2016.19.29.JPG","adapter":"local-images"},{"type":"file","name":"Wiretap.jpeg","path":"local-images:\/Wiretap.jpeg","extension":"jpeg","size":63826,"mime_type":"image\/png","width":1920,"height":1080,"create_date":"2024-04-05T22:22:01+00:00","create_date_formatted":"2024-04-05 22:22","modified_date":"2024-04-05T22:22:01+00:00","modified_date_formatted":"2024-04-05 22:22","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/Wiretap.jpeg","adapter":"local-images"},{"type":"file","name":"YourLogo.png","path":"local-images:\/YourLogo.png","extension":"png","size":2244,"mime_type":"image\/png","width":140,"height":29,"create_date":"2024-04-14T07:40:52+00:00","create_date_formatted":"2024-04-14 07:40","modified_date":"2024-04-14T07:40:52+00:00","modified_date_formatted":"2024-04-14 07:40","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/YourLogo.png","adapter":"local-images"},{"type":"file","name":"chrome_ImYyceKZUy.png","path":"local-images:\/chrome_ImYyceKZUy.png","extension":"png","size":22890,"mime_type":"image\/png","width":294,"height":297,"create_date":"2024-04-14T07:51:02+00:00","create_date_formatted":"2024-04-14 07:51","modified_date":"2024-04-14T07:51:02+00:00","modified_date_formatted":"2024-04-14 07:51","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/chrome_ImYyceKZUy.png","adapter":"local-images"},{"type":"file","name":"chrome_qazchOaIJp.png","path":"local-images:\/chrome_qazchOaIJp.png","extension":"png","size":268412,"mime_type":"image\/png","width":765,"height":617,"create_date":"2024-04-13T19:44:40+00:00","create_date_formatted":"2024-04-13 19:44","modified_date":"2024-04-13T19:44:40+00:00","modified_date_formatted":"2024-04-13 19:44","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/chrome_qazchOaIJp.png","adapter":"local-images"},{"type":"file","name":"estatement.pdf","path":"local-images:\/estatement.pdf","extension":"pdf","size":101968,"mime_type":"application\/pdf","width":0,"height":0,"create_date":"2024-08-12T11:17:41+00:00","create_date_formatted":"2024-08-12 11:17","modified_date":"2024-08-12T11:17:41+00:00","modified_date_formatted":"2024-08-12 11:17","adapter":"local-images"},{"type":"file","name":"joomla_black.png","path":"local-images:\/joomla_black.png","extension":"png","size":4979,"mime_type":"image\/png","width":225,"height":50,"create_date":"2024-02-26T15:19:51+00:00","create_date_formatted":"2024-02-26 15:19","modified_date":"2024-02-26T15:19:51+00:00","modified_date_formatted":"2024-02-26 15:19","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/joomla_black.png","adapter":"local-images"},{"type":"file","name":"ov002_02.png","path":"local-images:\/ov002_02.png","extension":"png","size":560437,"mime_type":"image\/png","width":419,"height":603,"create_date":"2024-08-09T11:47:38+00:00","create_date_formatted":"2024-08-09 11:47","modified_date":"2024-08-09T11:47:38+00:00","modified_date_formatted":"2024-08-09 11:47","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/ov002_02.png","adapter":"local-images"},{"type":"file","name":"ov002_03.png","path":"local-images:\/ov002_03.png","extension":"png","size":204400,"mime_type":"image\/png","width":402,"height":592,"create_date":"2024-08-09T12:00:12+00:00","create_date_formatted":"2024-08-09 12:00","modified_date":"2024-08-09T12:00:12+00:00","modified_date_formatted":"2024-08-09 12:00","thumb_path":"https:\/\/j51.test\/imagesD:\/repos\/j51\/images\/ov002_03.png","adapter":"local-images"},{"type":"file","name":"powered_by.png","path":"local-images:\/powered_by.png","extension":"png","size":3197,"mime_type":"image\/png","width":294,"height":44,"create_date":"2024-02-26T15:19:51+00:00","create_date_formatted":"2024-02-26 15:19","modified_date":"2024-02-26T15:19:51+00:00","modified_date_formatted":"2024-02-26 15:19","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/powered_by.png","adapter":"local-images"},{"type":"file","name":"screenshot1.png","path":"local-images:\/screenshot1.png","extension":"png","size":829880,"mime_type":"image\/png","width":1920,"height":1080,"create_date":"2024-09-04T10:57:48+00:00","create_date_formatted":"2024-09-04 10:57","modified_date":"2024-09-04T10:57:48+00:00","modified_date_formatted":"2024-09-04 10:57","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/screenshot1.png","adapter":"local-images"},{"type":"file","name":"sunset_logo.webp","path":"local-images:\/sunset_logo.webp","extension":"webp","size":9196,"mime_type":"image\/webp","width":320,"height":100,"create_date":"2024-07-31T09:45:45+00:00","create_date_formatted":"2024-07-31 09:45","modified_date":"2024-07-31T09:45:45+00:00","modified_date_formatted":"2024-07-31 09:45","thumb_path":"https:\/\/j51.test\/media\/cache\/com_media\/thumbs\/images\/sunset_logo.webp","adapter":"local-images"}]}

```

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on a9af0b603609fc67509c4d1799f2781216e95f5a<br /><br />my error was unrelated to this<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by brianteeman** [comment]
The file that broke is attached as a zip so it doesnt get mangled by github

[ov002_03.zip](https://github.com/user-attachments/files/17648044/ov002_03.zip)

**Comment by richard67** [comment]
@MacJoom Does this PR solve your issue #44417 ? If yes, please close the issue with reference to this PR. Thanks in advance.

**Comment by richard67** [comment]
@MacJoom Unit tests are failing:
```
There was 1 failure:

1) Joomla\Tests\Unit\Libraries\Cms\Image\ImageTest::testResizeTransparent
Failed asserting that false is true.

/drone/src/tests/Unit/Libraries/Cms/Image/ImageTest.php:420

```

**Comment by MacJoom** [comment]
I see, working on it.

**Comment by crimle** [comment]
I have tested this item :white_check_mark: successfully on a9af0b603609fc67509c4d1799f2781216e95f5a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on d4fdc54bf4f59a800e0eb9bb3a526b0edf9f713a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on d4fdc54bf4f59a800e0eb9bb3a526b0edf9f713a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

File: libraries/src/Image/Image.php [hunk 1]
```diff
@@ -435,12 +435,30 @@ public function crop($width, $height, $left = null, $top = null, $createNew = tr
...
-            $rgba  = imagecolorsforindex($this->getHandle(), imagecolortransparent($this->getHandle()));
-            $color = imagecolorallocatealpha($handle, $rgba['red'], $rgba['green'], $rgba['blue'], $rgba['alpha']);
-
-            // Set the transparent color values for the new image.
-            imagecolortransparent($handle, $color);
-            imagefill($handle, 0, 0, $color);
+            $ict  = imagecolortransparent($this->getHandle());
+            $ctot = imagecolorstotal($this->getHandle());
+            // Sanitize imagecolortransparent & imagecolorstotal
+            if ($ctot == 255 && $ict == 255) {
```
Comment (Quy) [inline libraries/src/Image/Image.php]: Do strict comparison.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 175a39f6003bad1ce784ff146c7f7d8c0be3ba01<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by richard67** [comment]
@MacJoom I've restored @Quy 's test result in the issue tracker which you have invalidated by a branch update. Please when you do that next time, updating a branch which has successful human tests, use the "Alter test" button in the issue tracker to restore the results. Thanks in advance.

**Comment by richard67** [comment]
I've also restored @brianteeman 's and @fgsw 's test result as the change after that was only for strict (typesafe) comparison, which does not cause a functional change as both of the called methods `imagecolortransparent` and `imagecolorstotal` can return only integers and so the compared variables can only be integers.

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44418">issues.joomla.org/tracker/joomla-cms/44418</a>.</sub>

**Comment by pe7er** [comment]
Thanks @MacJoom !

**Pull request merged**